assembly/src/lib.rs

```diff
@@ -1,50 +1,32 @@
 #![no_std]
-#![cfg_attr(all(nightly, not(feature = "std")), feature(error_in_core))]
 
 #[macro_use]
 extern crate alloc;
 
 #[cfg(any(test, feature = "std"))]
 extern crate std;
 
-use vm_core::{
-    Felt, ONE, Word, ZERO, prettier,
-    utils::{
-        ByteReader, ByteWriter, Deserializable, DeserializationError, DisplayHex, Serializable,
-    },
-};
+use miden_assembly_syntax as syntax;
+use miden_core::{ONE, ZERO};
 
 mod assembler;
-pub mod ast;
-pub mod diagnostics;
-mod library;
-mod parse;
-mod parser;
-mod sema;
 #[cfg(any(test, feature = "testing"))]
 pub mod testing;
 #[cfg(test)]
 mod tests;
 
 // Re-exported for downstream crates
 
-/// Merkelized abstract syntax tree (MAST) components defining Miden VM programs.
-pub use vm_core::mast;
-pub use vm_core::utils;
-
-pub use self::{
-    assembler::{Assembler, LinkLibraryKind, LinkerError},
-    diagnostics::{
-        DefaultSourceManager, Report, SourceFile, SourceId, SourceManager, SourceSpan, Span,
-        Spanned,
-    },
-    library::{
-        KernelLibrary, Library, LibraryError, LibraryNamespace, LibraryPath, LibraryPathComponent,
-        PathError, Version, VersionError,
-    },
-    parse::{Parse, ParseOptions},
-    parser::ModuleParser,
+pub use miden_assembly_syntax::{
+    DefaultSourceManager, KernelLibrary, Library, LibraryNamespace, LibraryPath, ModuleParser,
+    Parse, ParseOptions, Report, SourceFile, SourceId, SourceManager, SourceSpan, Span, Spanned,
+    ast, diagnostics, library, report,
 };
+/// Syntax components for the Miden Assembly AST
+/// Merkelized abstract syntax tree (MAST) components defining Miden VM programs.
+pub use miden_core::{mast, utils};
+
+pub use self::assembler::{Assembler, LinkLibraryKind, LinkerError};
```

Comment:
Probably not for this PR, but I wonder if we actually need the `assembler` namespace. Right now, the crate has basically just 3 files at the root level and everything else goes into the `assembler` directory.

Reply:
I had the same thought actually, but figured it would just add more noise to the changeset. Totally makes sense to me to flatten the namespacing internal to the `miden-assembly` crate though.

Reply:
I've gone ahead and done this in the last commit